### PR TITLE
Update build number of effectivet3

### PR DESCRIPTION
**Pull request by liaochenlanruo**
Update build number of effectivet3 to support python 3.7

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by liaochenlanruo** [comment]
@BiocondaBot please fetch artifacts

**Comment by liaochenlanruo** [comment]
@BiocondaBot please fetch artifacts

**Comment by liaochenlanruo** [comment]
@BiocondaBot please update

**Comment by liaochenlanruo** [comment]
@BiocondaBot please fetch artifacts

**Comment by liaochenlanruo** [comment]
@BiocondaBot please add label

**Comment by dpryan79** [comment]
Can you fix the wrapper script?

**Comment by liaochenlanruo** [comment]
> Can you fix the wrapper script?

I have asked @peterjc for help.

**Review by peterjc — CHANGES_REQUESTED** [review]
Your PR talks about python 3.7, but the change says python 2.7 onwards.

I am unclear why you need to specify this at all - I would assume that any Python will work (including 2.7 or 3.7)

**Comment by peterjc** [comment]
What problem are you trying to solve with this PR?

**Comment by peterjc** [comment]
I think the changes on the PR are currently meaningless, but the build is failing anyway,

```
12:41:20 BIOCONDA INFO (OUT) + effectivet3 -f empty.fasta -m TTSS_ANIMAL-1.0.1.jar -t selective -q
12:41:20 BIOCONDA INFO (OUT) Error: Unable to access jarfile /opt/conda/conda-bld/effectivet3_1645101122413/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/effectivet3-1.0.1-2/TTSS_GUI-1.0.1.jar
12:41:22 BIOCONDA INFO (OUT) Tests failed for effectivet3-1.0.1-h4a94de4_2.tar.bz2 - moving package to /opt/conda/conda-bld/broken
```

This is probably why the recipe testing got black listed some years back in 0782c3966677760a345f3dd750a9d3ed44735177 (10 Jul 2018), which I was unaware of until now

```
# unable to access jarfile
recipes/effectivet3
```

So it looks to me like something had broken the recipe long before the Python 2 to 3 transition, which I still think is irrelevant here.

My best guess is a path change somewhere in BioConda / conda which needs to be addressed in the recipe build script?

**Comment by dpryan79** [comment]
Yeah, It's looking for some `TTSS_GUI-1.0.1.jar` in some special place. I expect either that path or the jar file name have changed over the years. Anyway, I agree that there's no python restriction here, it should work with any version.

**Comment by peterjc** [comment]
Well spotted, yes - the JAR name changed when we sapped the URL from the original http://effectors.csb.univie.ac.at/sites/eff/files/others/TTSS_GUI-1.0.1.jar to the long term archived copy at https://depot.galaxyproject.org/software/TTSS_GUI/TTSS_GUI_1.0.1_src_all.jar

**Comment by dpryan79** [comment]
How the heck are the tests passing! They should be failing, I'll fix that.

File: recipes/effectivet3/effectivet3.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def main():
 
     java_args = [java] + mem_opts + prop_opts + [jar_arg] + [jar_path] + pass_args
 
-    sys.exit(subprocess.call(java_args))
+    sys.exit(subprocess.check_call(java_args))
```
Comment (peterjc) [inline recipes/effectivet3/effectivet3.py:98]: Why this change? Won't ``subprocess.call`` just return a non-zero on error, and that 
 return code whatever it is directly passed to the caller of the script via ``sys.exit``

https://docs.python.org/3/library/subprocess.html#subprocess.call

Comment (dpryan79) [reply recipes/effectivet3/effectivet3.py:98]: Hmm, could be, in which case java isn't returning a non-zero exit on errors...which is very problematic.

Comment (dpryan79) [reply recipes/effectivet3/effectivet3.py:98]: Ah, I see why these are failing, the tests need to know where these other jar files are.

Comment (dpryan79) [reply recipes/effectivet3/effectivet3.py:98]: Should the wrapper script handle the `-m something.jar` and convert the jar file path?

**Comment by liaochenlanruo** [comment]
> I think the changes on the PR are currently meaningless, but the build is failing anyway,
> 
> ```
> 12:41:20 BIOCONDA INFO (OUT) + effectivet3 -f empty.fasta -m TTSS_ANIMAL-1.0.1.jar -t selective -q
> 12:41:20 BIOCONDA INFO (OUT) Error: Unable to access jarfile /opt/conda/conda-bld/effectivet3_1645101122413/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/effectivet3-1.0.1-2/TTSS_GUI-1.0.1.jar
> 12:41:22 BIOCONDA INFO (OUT) Tests failed for effectivet3-1.0.1-h4a94de4_2.tar.bz2 - moving package to /opt/conda/conda-bld/broken
> ```
> 
> This is probably why the recipe testing got black listed some years back in [0782c39](https://github.com/bioconda/bioconda-recipes/commit/0782c3966677760a345f3dd750a9d3ed44735177) (10 Jul 2018), which I was unaware of until now
> 
> ```
> # unable to access jarfile
> recipes/effectivet3
> ```
> 
> So it looks to me like something had broken the recipe long before the Python 2 to 3 transition, which I still think is irrelevant here.
> 
> My best guess is a path change somewhere in BioConda / conda which needs to be addressed in the recipe build script?

Although the version of Python was not limited in the previous recipe, the version of Python displayed on the Bioconda page is equal to 2.7* （See below）. So, I changed `- python` to `- python >=2.7` in the recipe.
Recipe

However, when I search effectivet3 with mamba/conda, there found three builds (see below). 
Builds

When I try to install effectivet3 in a python 3.7 environment, the error happened.
`- package effectivet3-1.0.1-py36_0 requires python 3.6*, but none of the providers can be installed`

That's why I modified the recipe, and the test failed as you can see.

**Comment by dpryan79** [comment]
Ooof, yeah the current recipe doesn't really match was was built long ago. It seems those weren't noarch: python (maybe they predated that).

File: recipes/effectivet3/effectivet3.py (hunk 1 above)
Comment (dpryan79) [reply recipes/effectivet3/effectivet3.py:98]: Changing the path in `-m` won't actually do anything, since the TTSS_GUI jar file will prepend ./module to it. The only way I can see to get this to work is to either actually run it from within the directory containing the TTSS_GUI jar file or to symlink the module subdirectory into the current working directory. Both of these are kind of crappy options. Symlinking modules and then removing the symlink in the script is probably the simplest work-around, but it's prone to breaking if you have multiple runs in parallel in the same directory. Changing into the directory with TTSS_GUI.jar and then having the wrapper script translate the input and output file paths (or configuration file path) is probably the better option. Ideally the package authors would allow providing a path to the modules directory, but I assume this is no longer maintained since there hasn't been a new version released in years.

**Comment by peterjc** [comment]
As to the tool API, yes I agree the ``-m model.jar`` thing was clunky. But this is indeed a defunct tool still of niche interest.

Changing to the tool's directory was exactly how I was getting this to work once installed from the bioconda package - see https://github.com/peterjc/pico_galaxy/blob/master/tools/effectiveT3/effectiveT3.py (with matching Galaxy XML file).

I can see the argument for moving that path changing logic into the BioConda wrapper itself, but at the time I think I found packaging it hard enough and didn't want to further complicate the JAR wrapper.

Does BioConda have a standardised JAR wrapper now?
